django/views/generic/edit.py

```diff
@@ -225,12 +225,26 @@ def get_success_url(self):
                 "No URL to redirect to. Provide a success_url.")
 
 
-class BaseDeleteView(DeletionMixin, BaseDetailView):
+class BaseDeleteView(DeletionMixin, FormMixin, BaseDetailView):
     """
     Base view for deleting an object.
 
     Using this base class requires subclassing to provide a response mixin.
     """
+    form_class = Form
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        form = self.get_form()
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
```

Comment:
Custom post required for the form handling: 

- self.object must be populated before get_form(). (That would be BaseDetailView)
- But, DeletionMixin must be first base for get_success_url().
- So cannot use super().post() (for which we would use ProcessFormView).

Reply:
It's quite complicated. Maybe we can add a comment for the "future us" :older_adult: .